rotkehlchen/tasks/manager.py

```diff
@@ -547,13 +557,20 @@ def _schedule(self) -> None:
         if not_proceed:
             return  # too busy
 
-        callables = random.sample(
-            population=self.potential_tasks,
-            k=min(self.max_tasks_num - current_greenlets, len(self.potential_tasks)),
-        )
+        random.shuffle(self.potential_tasks)
```

Comment:
I see why you made this change. I actually meant to keep the previous functionality of the random sample and from that sample, if something is already running, just skip it.

Periodic task runs quite often anyway so even if we don't always have `max_tasks` running it would have been fine.

Plus it's more readable. The change with the for loop and reshuffling is a bit uglier. But I don't think that it's so bad as to insist on changing, so let's keep it for now. 

The positive (and I suppose why you chose it) is that if a task is already running and we have space for 1 more task, we don't lose the scheduling slot and will just go to the next available one.